### DISCUSSION TITLE
How to ignore the .js in favor of the .ts/.tsx?

### DISCUSSION THREAD

**Discussion by ghost** [preactjs/wmr#638] [Q&A] [answered]
I compile the files into the same directory where the sources are. As a result, both files (ts and js) lie side by side. wmr prefers the js. Is it possible to change the priority to the ts/tsx?

**Comment by developit** [comment, marked as answer]
Interesting... I'm curious what your use-case for compiling into the source directory is - that seems like it would cause a lot of problems beyond js/ts resolution priority, like potentially infinite filesystem watcher updates.

I think if you specify the extension in your import (`import './foo.ts'`) we will always use that file. TypeScript may complain, but they're going to have to rethink their position on file extensions anyway.
